app/forms/registration_form.rb

```diff
@@ -37,6 +40,7 @@ def persisted?
     property :barcode, virtual: true
     validates :source_id, format: { with: /\A.+:.+\z/, message: 'ID is invalid' }
     validates :catkey, allow_blank: true, format: { with: /\A\d+(:\d+)*\z/, message: 'is invalid' }
+    validates :barcode, allow_blank: true, format: { with: VALID_BARCODE_REGEX, message: 'is invalid' }
```

Comment:
This validation also needs to be client-side.

Reply:
yup - got it.